drivers/flow.c

```diff
@@ -0,0 +1,202 @@
+	uart_int_register_write_stream(uart_int_get_uart_handle(UID), &(flow->uart_stream_out));
+	
+	// Init MAVLink stream
+	mavlink_stream_conf_t mavlink_stream_conf =
+	{
+		.sysid       = 1,
+		.compid      = 50,
+		.debug       = true,
+	};
+	mavlink_stream_init(	&(flow->mavlink_stream),
+				&mavlink_stream_conf,
+				&(flow->uart_stream_in),
+				&(flow->uart_stream_out));
+}
+
+
+void flow_update(flow_t* flow)
+{
+	// Update time
+	flow->last_update_us  = time_keeper_get_micros();
+
+	// Receive incoming bytes
+	mavlink_stream_receive(&flow->mavlink_stream);
+
+	// Check if a new message is here
+	if( flow->mavlink_stream.msg_available == true )
+	{
+		// Get pointer to new message
+		mavlink_message_t* msg = &flow->mavlink_stream.rec.msg; 
+		
+		// Indicate that the message was handled
+		flow->mavlink_stream.msg_available = false;
+		
+		// declare messages
+		mavlink_optical_flow_t 			optical_flow_msg;
+		mavlink_data_transmission_handshake_t 	handshake_msg;
+		mavlink_encapsulated_data_t		data_msg;
+
+		switch( msg->msgid )
+		{
+			case MAVLINK_MSG_ID_OPTICAL_FLOW:
+				mavlink_msg_optical_flow_decode(msg, &optical_flow_msg);
+				
+				flow->tmp_flow_x 	= optical_flow_msg.flow_x;
+				flow->tmp_flow_y 	= optical_flow_msg.flow_y;
+				flow->tmp_flow_comp_m_x = optical_flow_msg.flow_comp_m_x;
+				flow->tmp_flow_comp_m_y = optical_flow_msg.flow_comp_m_y;
+			break;
+			case MAVLINK_MSG_ID_DATA_TRANSMISSION_HANDSHAKE:
+				mavlink_msg_data_transmission_handshake_decode(msg, &handshake_msg);
+				// Get type of handshake
+				flow->handshake_state = handshake_msg.jpg_quality;
+				
+				// Get number of of vectors
+				flow->of_count 	      = handshake_msg.width;
+				if( flow->of_count > 125 )
+				{
+					flow->of_count = 125;
+				}
+
+				// Get total payload size
+				flow->n_packets	      = handshake_msg.packets;
+				flow->size_data	      = handshake_msg.size;
+				if( flow->size_data > 500 )
+				{
+					flow->size_data = 500;
+				}
+			break;
+			case MAVLINK_MSG_ID_ENCAPSULATED_DATA:
+				mavlink_msg_encapsulated_data_decode(msg, &data_msg);
+				switch( flow->handshake_state )
+				{
+					case FLOW_HANDSHAKE_METADATA:
+						if( data_msg.seqnr < flow->n_packets - 1 )
+						{
+							// not last packet
+							for ( int i = 0; i < MAVLINK_MSG_ENCAPSULATED_DATA_FIELD_DATA_LEN; ++i)
+							{
+								flow->of_loc.data[i + data_msg.seqnr * MAVLINK_MSG_ENCAPSULATED_DATA_FIELD_DATA_LEN] = data_msg.data[i];
+							}
+						}
+						else if( data_msg.seqnr < flow->n_packets )
+						{
+							// last packet
+							for ( int i = 0; i < flow->size_data; ++i)
+							{
+								flow->of_loc.data[i + data_msg.seqnr * MAVLINK_MSG_ENCAPSULATED_DATA_FIELD_DATA_LEN] = data_msg.data[i];
+							}
+
+							// swap bytes
+							for (int i = 0; i < flow->of_count; ++i)
+							{
+								flow->of_loc.x[i] = endian_rev16(flow->of_loc.x[i]);
+								flow->of_loc.y[i] = endian_rev16(flow->of_loc.y[i]);
+							}
+						}
+					break;
```

Comment:
the formatting we agreed was: a line return after the break, except for the last one, could you reformat this in these switch statement?